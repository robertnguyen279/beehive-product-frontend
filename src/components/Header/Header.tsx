import React from 'react';

export default function Header(): React.ReactElement {
  return <div>Header</div>;
}
