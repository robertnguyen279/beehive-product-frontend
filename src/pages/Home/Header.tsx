import React from 'react';
import { HomeChild } from 'types/main';
import Logo from 'public/icons/white-logo.svg';
import Cart from 'public/icons/shopping-cart.svg';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import { toggleMenuSldier } from 'actions/ui';
import { RootState } from 'reducers';

const HeaderHome = ({ isLaptop }: HomeChild): React.ReactElement => {
  const { t } = useTranslation('HomePage');
  const dispatch = useDispatch();
  const openMenuSlider = useSelector((state: RootState) => state.ui.openMenuSlider);

  const handleToggleMenuSlider = (): void => {
    dispatch(toggleMenuSldier());
  };

  console.log(openMenuSlider);

  const setBarClassName = (bar: number): string => {
    if (openMenuSlider === undefined) {
      return `home__header__right__slider-button__bar-${bar}`;
    } else if (openMenuSlider === false) {
      return `home__header__right__slider-button__bar-${bar}--inactive`;
    } else {
      return `home__header__right__slider-button__bar-${bar}--active`;
    }
  };

  return isLaptop ? (
    <div className="home__header z-10 relative xl:h-32 lg:h-24 md:h-20 m-auto flex justify-between items-center">
      <div className="home__header__logo">
        <img src={Logo} alt="logo" className="w-30" />
      </div>
      <div className="home__header__right flex items-center text-white text-base justify-start">
        <div className="home__header__right__cart w-8 bg-gray-600 h-8 flex justify-center items-center cursor-pointer">
          <img src={Cart} alt="shopping-cart" className="w-4/6" />
        </div>
        <div className="home__header__right__login mx-4">{t('login-button')}</div>
        <div className="home__header__right__slider-button cursor-pointer relative" onClick={handleToggleMenuSlider}>
          <div className={setBarClassName(1)}>
            <span>.</span>
          </div>
          <div className={setBarClassName(2)}>
            <span>.</span>
          </div>
          <div className={setBarClassName(3)}>
            <span>.</span>
          </div>
        </div>
      </div>
    </div>
  ) : (
    <div>mobile</div>
  );
};

export default HeaderHome;
