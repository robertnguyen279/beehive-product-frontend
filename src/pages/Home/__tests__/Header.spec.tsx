import React from 'react';
import { Provider } from 'react-redux';
import Header from '../Header';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
// const clickFn = jest.fn();
const mockStore = configureStore([]);
const store = mockStore({});

describe('Header', () => {
  const component = shallow(
    <Provider store={store}>
      <Header />
    </Provider>,
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
