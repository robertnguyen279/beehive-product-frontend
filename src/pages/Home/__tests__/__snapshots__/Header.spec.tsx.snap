// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="home__header z-10 relative xl:h-32 lg:h-24 h-20 m-auto flex justify-between items-center md:py-7 py-5"
  id="login-header"
>
  <div
    className="home__header__logo w-52 px-5 cursor-pointer"
  >
    <img
      alt="logo"
      src="test-file-stub"
    />
  </div>
  <div
    className="home__header__right flex items-center text-white text-base justify-start mr-4"
  >
    <div
      className="home__header__right__cart w-8 bg-gray-600 h-8 flex justify-center items-center cursor-pointer"
    >
      <img
        alt="shopping-cart"
        className="w-4/6"
        src="test-file-stub"
      />
    </div>
    <div
      className="home__header__right__login mx-4 cursor-pointer"
    >
      <CountrySelector />
    </div>
    <div
      className="home__header__right__slider-button cursor-pointer relative"
      onClick={[Function]}
    >
      <div
        className="home__header__right__slider-button__bar-1"
      >
        <span>
          .
        </span>
      </div>
      <div
        className="home__header__right__slider-button__bar-2"
      >
        <span>
          .
        </span>
      </div>
      <div
        className="home__header__right__slider-button__bar-3"
      >
        <span>
          .
        </span>
      </div>
    </div>
  </div>
</div>
`;
